include/libassert/libassert_internal.hpp

```diff
@@ -0,0 +1,362 @@
+#ifndef LIBASSERT_INTERNAL_HPP
+#define LIBASSERT_INTERNAL_HPP
```

Comment:
Seeing the contents now, maybe something like `platform.hpp` would be best for the name. I'll give more thought to the split, I think there's likely no reason not to want to split into multiple headers now which means the monolothic assert.hpp can be broken down a bunch later, which is cool.

Reply:
I am partial to the multi file approach as its generally far easier to read over vs having a massive file.